qiita_db/analysis.py

```diff
@@ -753,7 +753,7 @@ def _add_file(self, filename, filetype, data_type=None, conn_handler=None):
                "checksum, checksum_algorithm_id, data_directory_id) VALUES "
                "(%s, %s, %s, %s, %s) RETURNING filepath_id")
         # magic number 1 is for crc32 checksum algorithm
-        fpid = conn_handler.execute_fetchone(sql, (fullpath, fptypeid,
+        fpid = conn_handler.execute_fetchone(sql, (filename, fptypeid,
```

Comment:
Ok, if you can use `insert_filepaths` to do part of the work here, use it, so we avoid code duplication. If you still need some code to finish other work it is fine to do it here.